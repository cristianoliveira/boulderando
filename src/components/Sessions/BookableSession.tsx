import React from 'react'
import { TableCell, TableCellProps, TableRow, Button } from '@mui/material'
import EventAvailableIcon from '@mui/icons-material/EventAvailable'

import getNextPossibleDay from '../../modules/weekday'

import { SESSION_LIST_TABLE_ITEM } from './data-testid'

const TbCell = (props: TableCellProps) => (
  <TableCell
    sx={{
      padding: '8px',
    }}
    {...props}
  />
)

type BookableSessionProps = {
  session: Session
  hasBeenBooked: boolean
  isProcessing: boolean
  onBookingSubmit(sessions: Session): void
}

const BookableSession: React.FC<BookableSessionProps> = ({
  session,
  hasBeenBooked,
  isProcessing,
  onBookingSubmit,
}) => (
  <TableRow
    data-testid={SESSION_LIST_TABLE_ITEM}
    sx={{
      color: hasBeenBooked ? 'gray' : 'green',
    }}
  >
    <TbCell>{session.gym_name}</TbCell>
    <TbCell>
      {session.day_of_week}-{getNextPossibleDay(session.day_of_week)}
    </TbCell>
    <TbCell>{session.time}</TbCell>
    <TbCell align={'right'}>
      <Button
        disabled={isProcessing || hasBeenBooked}
        variant="contained"
        type="submit"
        onClick={() => onBookingSubmit(session)}
      >
        <EventAvailableIcon />
      </Button>
    </TbCell>
  </TableRow>
)

export default BookableSession
