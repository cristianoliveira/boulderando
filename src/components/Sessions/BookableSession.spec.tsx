import React from 'react'
// eslint-disable-next-line
import renderer from 'react-test-renderer'

import BookableSession from './BookableSession'

const session = {
  gym_name: 'basement',
  day_of_week: 'saturday',
  scheduled_time: 'saturday',
  time: '10:30 - 12:30',
}

describe('Form', () => {
  it('matches the snapshot', async () => {
    expect(
      renderer
        .create(
          <BookableSession
            isProcessing={false}
            hasBeenBooked={false}
            session={session}
            onBookingSubmit={jest.fn()}
          />
        )
        .toJSON()
    ).toMatchSnapshot()
  })
})
