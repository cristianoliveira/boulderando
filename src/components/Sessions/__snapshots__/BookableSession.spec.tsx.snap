// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches the snapshot 1`] = `
<tr
  className="MuiTableRow-root css-o2zb70-MuiTableRow-root"
  data-testid="session_list_table_item"
  role={null}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-sizeMedium css-io405h-MuiTableCell-root"
  >
    basement
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-sizeMedium css-io405h-MuiTableCell-root"
  >
    saturday
    -
    21/05/2022
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-sizeMedium css-io405h-MuiTableCell-root"
  >
    10:30 - 12:30
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignRight MuiTableCell-sizeMedium css-11fwb4b-MuiTableCell-root"
  >
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="EventAvailableIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.53 11.06 15.47 10l-4.88 4.88-2.12-2.12-1.06 1.06L10.59 17l5.94-5.94zM19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11z"
        />
      </svg>
    </button>
  </td>
</tr>
`;
